pkg/jaegerquery/find_traces.go

```diff
@@ -0,0 +1,71 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package jaegerquery
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	jaegertranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+	"github.com/timescale/promscale/pkg/pgxconn"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+func findTraces(ctx context.Context, conn pgxconn.PgxConn, q *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
+	query, params := findTracesQuery(q)
+	rows, err := conn.Query(ctx, query, params...)
+	if err != nil {
+		return nil, fmt.Errorf("querying traces: %w", err)
+	}
+	defer rows.Close()
+
+	return scanTraces(rows)
+}
+
+func scanTraces(rows pgxconn.PgxRows) ([]*model.Trace, error) {
+	traces := pdata.NewTraces()
+	for rows.Next() {
+		if rows.Err() != nil {
+			return nil, fmt.Errorf("trace row iterator: %w", rows.Err())
+		}
+		ScanRow(rows, &traces)
```

Comment:
1. I agree with the bot
2. I don't think passing ref of `traces` is required, since internally, `orig` already has a ref. We can remove the ref here.

Reply:
Fixed the error. Keeping as a reference in case the address orig points to needs to change.